### DISCUSSION TITLE
DBeaver tries to write to a SQLite read-only dabase connection (open_mode 1)

### DISCUSSION THREAD

**Discussion by yuuyins** [dbeaver/dbeaver#16514] [Q&A]
## Happened
Try listing tables in a read only SQLite database connection

```command
SQL Error [255]: [UNKNOWN_ERROR] unknown error (attempt to write a readonly database)
  [UNKNOWN_ERROR] unknown error (attempt to write a readonly database)
  [UNKNOWN_ERROR] unknown error (attempt to write a readonly database)
```

That is, solution given in https://github.com/dbeaver/dbeaver/discussions/148 by @serge-rider won't work further than for connecting, because it seems to try writing to the database afterwards...

https://user-images.githubusercontent.com/86538850/168877220-b70f448c-56c2-41e5-8408-868341796b67.mp4

## Reproduce
1. Connect with read-only SQLite databae (specifically a Nix database `/nix/var/nix/db/db.sqlite`)
2. Set driver parameter "open_mode" to "1" to make connection read-only
3. Ok

4. Try clicking tables

**Comment by LonwoLonwo** [comment]
Can you please share the full [error log](https://github.com/dbeaver/dbeaver/wiki/Log-files)?

**Reply by yuuyins** [reply]
[dbeaver.log](https://github.com/dbeaver/dbeaver/files/8730455/dbeaver.log)

dbeaver Version 22.0.2.202204131658
NixOS 22.05.20220415.5181d59 (Quokka) Linux 5.15.34 x86_64 x

I'm only able to reproduce the issue with `/nix/var/nix/db/db.sqlite` (`-rw-r--r-- 1 root root 95473664 May 18 21:57 /nix/var/nix/db/db.sqlite`). I was not able to reproduce the issue by creating myself a sqlite readonly database owned by root for example. I will ask the Nix people if there is something special going on...

**Reply by yuuyins** [reply]
so i was able to reproduce it @LonwoLonwo 

1. download [nixdb.zip](https://github.com/dbeaver/dbeaver/files/8730804/nixdb.zip) in a Linux distribution

2. `sudo unzip -X nixdb.zip`

3. check the permissions `ls -la nixdb`

       drwxr-xr-x 1 root root       18 May 19 12:16 .
       drwxr-xr-x 1 root root       28 May 19 12:31 ..
       -rw-r--r-- 1 root root 95473664 May 19 12:16 db.sqlite

4. Connect dbeave to nixdb/db.sqlite while using `open_mode 1`
